geode-core/src/main/java/org/apache/geode/internal/cache/EntryEventImpl.java

```diff
@@ -235,6 +237,13 @@ public void fromData(DataInput in) throws IOException, ClassNotFoundException {
   protected EntryEventImpl(InternalRegion region, Operation op, Object key, boolean originRemote,
       DistributedMember distributedMember, boolean generateCallbacks, boolean fromRILocalDestroy) {
     this.region = region;
+    InternalDistributedSystem ds =
+        (InternalDistributedSystem) region.getCache().getDistributedSystem();
+    if (ds.getOffHeapStore() != null) {
+      this.offHeapLock = new Object();
```

Comment:
It looks like we could end up initializing the offHeapLock more than one place, which seems like an issue to me. Shouldn't this only be initialized in the constructor (if needed) and then used everywhere else? Or, alternately, a check could be added here for if offHeapLock is already non-null. My concern is that we would be creating a new lock each time fromData is called (and it is called all over the place), and then for each call get a different lock object so that we wouldn't actually be locking, because each call would be the only one holding that particular lock.

Reply:
I am unclear how this is working given that offHeapLock is final. Is it actually only being set once?

Reply:
You are right about it being final. EntryEventImpl has multiple constructors and this is just one of them. I don't see fromData initializing offHeapLock. For an EntryEventImpl that we deserialize it will use the no-arg constructor which sets offHeapLock to null.

Reply:
You're right. The way it was showing the diff in Github made it look like that line was in fromData instead of in a constructor